File: Data Structures & Algorithms/max-water-container/submission-0.rs
impl Solution {
    pub fn max_area(heights: Vec<i32>) -> i32 {
        let (mut left, mut right) = (0usize, heights.len() - 1);
        let mut max_area = i32::MIN;
        while left < right {
            let left_wall = heights[left];
            let right_wall = heights[right];
            let area = (right - left) as i32 * left_wall.min(right_wall);
            if area > max_area {
                max_area = area;
            }
            if left_wall < right_wall {
                left += 1;
            } else {
                right -= 1;
            }
        }
        max_area
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        auto left{0}, right{static_cast<int>(nums.size()) - 1}, mid{0};
        while (left < right) {
            mid = (left + right) / 2;
            if (nums[right] > nums[mid]) {
                right = mid;
            } else  {
                left = mid + 1;
            }
        }
        return nums[left];
    }
};
